txscript/script.go

```diff
@@ -367,6 +377,44 @@ func (s *Script) checkHashTypeEncoding(hashType SigHashType) error {
 	return nil
 }
 
+func (s *Script) checkLowDERSignature(sig []byte) error {
```

Comment:
After reviewing more, I personally don't think this function should exist at all.  Rather, the `checkSignatureEncoding` function should check the flag at the appropriate place and do the check there:

```Go
if s.verifyLowS { /* do the checks and return error if no good */ }
```